source/slang/slang-check-conformance.cpp

```diff
@@ -104,6 +104,16 @@ namespace Slang
         // For now we are continuing to conflate all the subtype-ish relationships but not
         // tangling convertibility into it.
 
+        // First, make sure both sub type and super type decl are ready for lookup.
+        if (auto subDeclRefType = as<DeclRefType>(subType))
+        {
+            ensureDecl(subDeclRefType->declRef.getDecl(), DeclCheckState::ReadyForLookup);
+        }
```

Comment:
The call to `getInheritanceInfo()` below should end up doing an `ensureDecl()` in the case where the type is a `DeclRefType`, so I'm not 100% sure why this bit would be needed.